tables/TaQL/ExprLogicNode.cc

```diff
@@ -286,41 +288,53 @@ TableExprNodeINInt::TableExprNodeINInt (const TableExprNodeRep& node,
                                         Bool)
 : TableExprNodeBinary (NTBool, node, OtIN)
 {}
-void TableExprNodeINInt::convertConstChild()
+TableExprNodeINInt::~TableExprNodeINInt()
+{}
+void TableExprNodeINInt::optimize()
+{
+  doOptimize (rnode_p);
+}
+void TableExprNodeINInt::doOptimize (TENShPtr& rnode)
 {
-  if (rnode_p->isConstant()  &&  rnode_p->valueType() == VTArray) {
-    // Convert array to a set for lookup
-    MArray<Int64> values = rnode_p->getArrayInt(0);
+  if (rnode->isConstant()  &&  rnode->valueType() == VTArray) {
+    // Convert a constant array for faster lookup.
+    MArray<Int64> values = rnode->getArrayInt(0);
     Array<Int64> arr(values.array());
     if (values.hasMask()) {
       // Remove masked elements.
       arr.reference (values.flatten());
     }
-    if (! arr.empty()) {
-      itsIndexSet.clear();
-      itsIndexSet.insert(arr.begin(), arr.end());
-    }
+    // Use an unordered_set for fast lookup.
```

Comment:
It is using an unordered_map now, so changed the comment.
Faster than what for uInt64 keys? An unordered_map is certainly much faster than using class TableExprNodeSet itself. I've tried a Block<Bool> for integer values with a small range, but that was hardly any faster than unordered_map. The map was needed for large ranges, so I decided to only use maps.
It is using Int64 keys, since the integer values are signed.